File: Data Structure CDAC/DS_Sept_23/_03_Stack/src/CharStack/cStack.java
package CharStack;

public class cStack {
	int size,top;
	char arr[];
	public cStack()
	{
		size=3;
		top=-1;
		arr=new char[size];
	}
	public cStack(int size)
	{
		this.size=size;
		top=-1;
		arr=new char[size];
	}
	public boolean isEmpty()
	{
		if(top==-1)
			return true;
		else
			return false;
	}
	public boolean isFull()
	{
		return top==size-1?true:false;
	}
//		if(top==size-1)
//			return true;
//		else
//			return false;
//	}
	public void push(char ch)
	{
		if(!isFull())
		{
			arr[++top]=ch;
		}
		else
			System.out.println("Stack Is Full !!! ");
	}
	public char pop()
	{
		char ch='\0';
		if(!isEmpty())
			ch=arr[top--];
		return ch;
	}
	public void display()
	{
		System.out.println(" ---------- Stack Is ----------\n");
		for(int i=top;i>=0;i--)
			System.out.println("\t"+arr[i]);
	}
}

File: CPP Sept 2023/Templates_in_cpp/cStack.cpp
#include<iostream>
#include<string.h>
using namespace std;
template <class T>
class cStack{
    int size;
    int top;
    T *arr;
    public:
    
    cStack(){
        size = 3;
        top = -1;
        arr = new T[3];
    }
    cStack(int s){
        size = s;
        top = -1;
        arr = new T[size];
    }
    void push(int no){
        if(top<size){
            top++;
            arr[top]=no;
        }
        else{
            cout<<"Stack is overflow.";
        }
    }
    T pop(){
        if(top>=0){
            T no = arr[top];
            top--;
            return no;

        }
    }
    void display(){
        for(int i = top;i>=0;i--){
            cout<<arr[i]<<endl;
        }
    }
};

int main()
{
    int value;
    cStack <int> s1;
    
    int ch;
    

    do{
        cout<<"Enter the choice: 1. push()\
                          \n 2. pop(\
                          \n 3. display()\
                          \n 4. 0 to exit"<<endl;
        cin>>ch;
        switch (ch)
        {
        case 1:
        cout<<"Enter value: ";
        cin>>value;
        s1.push(value);
        
            break;
        
        case 2:
        s1.pop();
        break;

        case 3:
        cout<<"The stack is: "<<endl;
        s1.display();
        break;
        }
    }while(ch!=0);
    return 0;
  

}
